workspace/techsupport/jobs.py

```diff
@@ -117,47 +78,29 @@ def out_of_office_status():
         return f"Tech support out of office is currently ON until {end}."
 
 
-def report_rota():
-    rows = get_rota_data_from_sheet()
-    rota = {row[0]: (row[1], row[2]) for row in rows[1:] if len(row) >= 3}
-
-    blocks = [
-        {
-            "type": "header",
-            "text": {
-                "type": "plain_text",
-                "text": "Tech support rota",
-            },
-        },
-    ]
-
-    today = date.today()
-    this_monday = today - timedelta(days=today.weekday())
-    blocks.append(get_rota_block_for_week(rota, this_monday, this_or_next="this"))
-
-    next_monday = this_monday + timedelta(days=7)
-    blocks.append(get_rota_block_for_week(rota, next_monday, this_or_next="next"))
-
-    blocks.append(get_block_linking_rota_spreadsheet(tech_support_rota_spreadsheet_id))
-    return json.dumps(blocks, indent=2)
-
-
-def get_rota_data_from_sheet():  # pragma: no cover
-    credentials = service_account.Credentials.from_service_account_file(
-        environ["GCP_CREDENTIALS_PATH"],
-        scopes=["https://www.googleapis.com/auth/spreadsheets.readonly"],
-    )
-    service = discovery.build("sheets", "v4", credentials=credentials)
-
-    return (
-        service.spreadsheets()
-        .values()
-        .get(
-            spreadsheetId=tech_support_rota_spreadsheet_id,
-            range="Rota",
+class TechSupportRotaReporter(RotaReporter):
+    def __init__(self):
```

Comment:
You shouldn't need this `__init__` method here - you can just instantiate the class on line 103 with 
```
def report_rota():
    return TechSupportRotaReporter(
        title="Tech support rota",
        spreadsheet_id="1q6EzPQ9iG9Rb-VoYvylObhsJBckXuQdt3Y_pOGysxG8",
        sheet_range="Rota",
).report()
```

Reply:
I see! Will fix this and the others :)